### PR TITLE
feat(core): support dynamic assets path for icon

### PR DESCRIPTION
**Pull request by splincode**

### CODE REVIEW

File: projects/core/pipes/flag/flag.pipe.ts [hunk 1]
```diff
@@ -20,6 +20,11 @@ export class TuiFlagPipe implements PipeTransform {
             return null;
         }
 
-        return `${this.staticPath}/flags/${countryIsoCode.toLowerCase()}.svg`;
+        const baseUrl =
+            typeof this.staticPath === 'string'
+                ? this.staticPath
+                : this.staticPath(countryIsoCode);
+
+        return `${baseUrl}/flags/${countryIsoCode.toLowerCase()}.svg`;
```
Comment (nsbarsukov) [inline projects/core/pipes/flag/flag.pipe.ts:28]: nitpicking:
```suggestion
        const assetsPath =
            typeof this.staticPath === 'string'
                ? this.staticPath
                : this.staticPath(countryIsoCode);

        return `${assetsPath}/flags/${countryIsoCode.toLowerCase()}.svg`;
```

File: projects/core/tokens/assets-path.ts [hunk 2]
```diff
@@ -1,7 +1,9 @@
 import type {ValueProvider} from '@angular/core';
 import {tuiCreateToken} from '@taiga-ui/cdk/utils/miscellaneous';
 
-export const TUI_ASSETS_PATH = tuiCreateToken('assets/taiga-ui/icons');
+export const TUI_ASSETS_PATH = tuiCreateToken<string | ((icon: string) => string)>(
```
Comment (nsbarsukov) [inline projects/core/tokens/assets-path.ts:4]:
```suggestion
export const TUI_ASSETS_PATH = tuiCreateToken<string | TuiStringHandler<string>>(
```

File: projects/core/tokens/icon-resolver.ts [hunk 3]
```diff
@@ -11,10 +11,16 @@ export const TUI_ICON_START_RESOLVER = tuiCreateTokenFromFactory<
...
-            : `${path}/${icon.replace('@tui.', '').split('.').join('/')}.svg`;
+    return (icon) => {
+        if (!icon || icon.includes('/')) {
+            return icon;
+        }
+
+        const baseUrl = typeof path === 'string' ? path : path(icon);
+        const name = icon.replace('@tui.', '').split('.').join('/');
+
+        return `${baseUrl}/${name}.svg`;
```
Comment (nsbarsukov) [inline projects/core/tokens/icon-resolver.ts:22]: nitpicking:

```suggestion
        const assetsPath = typeof path === 'string' ? path : path(icon);
        const name = icon.replace('@tui.', '').split('.').join('/');

        return `${assetsPath}/${name}.svg`;
```

**Review by nsbarsukov — APPROVED** [review]

File: projects/core/tokens/assets-path.ts (hunk 2 above)
Comment (splincode) [reply projects/core/tokens/assets-path.ts:4]: Circular: cdk -> core -> cdk

File: projects/core/tokens/icon-resolver.ts (hunk 3 above)
Comment (splincode) [reply projects/core/tokens/icon-resolver.ts]: fixed

File: projects/core/pipes/flag/flag.pipe.ts (hunk 1 above)
Comment (splincode) [reply projects/core/pipes/flag/flag.pipe.ts]: fixed

**Pull request closed without merging**